src/main/java/study/racing/Car.java

```diff
@@ -0,0 +1,30 @@
+package study.racing;
+
+public class Car {
+
+  private int position = 0;
+
+  private static final int MOVE_CONDITION = 4;
+
+  public Car() { }
+
+  public Car(int position) {
+    this.position = position;
+  }
+
+  public int getPosition() {
+    return position;
+  }
+
+  public void moveByCondition(int condition) {
+    if (condition >= MOVE_CONDITION) {
+      move();
+    }
+  }
+
+  private void move() {
+    this.position++;
+  }
```

Comment:
```suggestion
  public void move(MoveStrategy moveStrategy) {
    if (moveStrategy.isMove()){
      this.position++;
    }
  }
```
이런식으로 작동할 수 있게 수행해보면 어떨까요?